### ISSUE TITLE
Warnings when building plperl with clang-12 and above

### ISSUE THREAD

**Issue by l-wang** [greenplum-db/gpdb#12735]
### Greenplum version or build
GPDB7

### OS version and uname -a
Linux station3 5.13.0-20-generic #20-Ubuntu SMP Fri Oct 15 14:21:35 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### autoconf options used ( config.status --config )

'--enable-depend' '--enable-debug' '--enable-cassert' '--with-perl' '--with-python' '--with-libxml' '--prefix' '/usr/local/gpdb' 'CC=clang-12' 'CXX=clang++-12' 'CFLAGS=-O0 -fno-omit-frame-pointer' 'CPPFLAGS=-O0 -fno-omit-frame-pointer'

### Expected behavior
Fewer (or none) warnings during compilation

### Actual behavior

120+ warnings similar to below:
```
plperl.c:2368:2: warning: '}' and ')' tokens terminating statement expression appear in different macro expansion contexts [-Wcompound-token-split-by-macro]
        PUSHMARK(sp);
        ^~~~~~~~~~~~
/usr/lib/x86_64-linux-gnu/perl/5.32/CORE/pp.h:68:5: note: expanded from macro 'PUSHMARK'
    } STMT_END
    ^
plperl.c:2368:2: note: ')' token is here
        PUSHMARK(sp);
        ^~~~~~~~~~~~
/usr/lib/x86_64-linux-gnu/perl/5.32/CORE/pp.h:68:7: note: expanded from macro 'PUSHMARK'
    } STMT_END
      ^~~~~~~~
./ppport.h:3827:20: note: expanded from macro 'STMT_END'
#  define STMT_END      )
```

The `-Wcompound-token-split-by-macro` warning was introduced in clang-12 by commit https://reviews.llvm.org/rG0e00a95b4fad5e72851de012d3a0b2c2d01f8685. Now that clang-13 is released, we better keep up-to-date and fix these warnings.

**Comment by beeender** [comment]
same issue on pg master has not been fixed yet.

**Comment by beeender** [comment]
Upstream fixed this issue by 4eec2e03c34f9a357d86317b1cd05d80a2ab559d 
And it should be backported.
